### ISSUE TITLE
Cache flush removes all items

### ISSUE THREAD

**Issue by bkirova** [phalcon/cphalcon#12153]
The flush function in Phalcon\Cache\Backend\Apc removes all elements in apc even if there are items with a different prefix.

Phalcon Code :
```
for item in iterator(new \APCIterator("user")) {
     apc_delete(item["key"]);
}
```

**It would be nice if a single instance of the cache removes only it's own keys by prefix or at least if there was an optional parameter prefix that select the elements with that prefix and removes only those.**

### Possible usage of the prefix variable

```
		public function flush() -> boolean
		{
			var item, prefixPattern;
			
			let prefixPattern = "/^_PHCA" . this->_prefix . "/";
	
			for item in iterator(new \APCIterator("user", prefixPattern)) {
				apc_delete(item["key"]);
			}
	
			return true;
		}
```

**Comment by Jurigag** [comment]
Well in 3.1.0 there will be added APCu backends, so maybe add this too then @sergeyklay ?

**Comment by sergeyklay** [comment]
@Jurigag This might be fixed with preserving backward compatibility

**Comment by sergeyklay** [comment]
@bkirova I'll fix it ASAP

**Comment by sergeyklay** [comment]
Fixed in the `3.0.x` branch.

```sh
git clone git@github.com:phalcon/cphalcon.git
cd cphalcon
git checkout 3.0.x

zephir fullclean
zephir build
```
